report a misleading "401 Unauthorized"

<!-- Thanks for filing an issue! Please use this template to provide more context for the issue . -->

**Description of the issue**:
report a "401 Unauthorized", but point to 'private-docker.changhongedu.cn/product-upgrade'

**Expected behavior**:
`com.google.cloud.tools.jib.registry.RegistryUnauthorizedException: Unauthorized for private-docker.changhongedu.cn/product-upgrade`
this line should be
`com.google.cloud.tools.jib.registry.RegistryUnauthorizedException: Unauthorized for hub.c.163.com/azul/zulu-openjdk:8`

**Environment**: <!-- OS, Maven, and other relevant environment information -->
windows 10
maven 3.5.4
jib-maven-plugin 0.9.10

**`jib-maven-plugin` Configuration**: <!-- Delete this section if not used -->
```xml
<from>
  <image>hub.c.163.com/azul/zulu-openjdk:8</image>
</from>
<to>
  <image>private-docker.changhongedu.cn/product-upgrade</image>
</to>
```

**Log output**: <!-- If applicable, provide relevant log output -->
```

[DEBUG] TIMING  Building and pushing image
[DEBUG] RUNNING Building and pushing image
[INFO] Retrieving registry credentials for private-docker.changhongedu.cn...
[DEBUG] TIMING  Retrieving registry credentials for private-docker.changhongedu.cn
[DEBUG] RUNNING Retrieving registry credentials for private-docker.changhongedu.cn
[INFO] Getting base image hub.c.163.com/azul/zulu-openjdk:8...
[DEBUG] TIMING  Building application layers
[DEBUG] TIMING  Pulling base image manifest
[DEBUG] Using Maven settings for private-docker.changhongedu.cn
[DEBUG] RUNNING Pulling base image manifest
[DEBUG] RUNNING Building application layers
[DEBUG] TIMED   Retrieving registry credentials for private-docker.changhongedu.cn : 22.613 ms
[INFO] Building classes layer...
[DEBUG] TIMED   Building application layers : 21.77 ms
[INFO] Building resources layer...
[INFO] Building snapshot dependencies layer...
[INFO] Building dependencies layer...
[DEBUG] TIMING  Building snapshot dependencies layer
[DEBUG] TIMING  Building resources layer
[DEBUG] Building application layers : 21.77 ms
[DEBUG] TIMING  Building classes layer
[DEBUG] RUNNING Building resources layer
[DEBUG] RUNNING Building snapshot dependencies layer
[DEBUG] TIMING  Building dependencies layer
[DEBUG] TIMING  Setting up to push layers
[DEBUG] Retrieving registry credentials for private-docker.changhongedu.cn : 22.613 ms
[DEBUG] RUNNING Building classes layer
[DEBUG] TIMING  Authenticating with push to private-docker.changhongedu.cn
[DEBUG] RUNNING Setting up to push layers
[DEBUG] RUNNING Building dependencies layer
[DEBUG] TIMED   Setting up to push layers : 20.982 ms
[DEBUG] RUNNING Authenticating with push to private-docker.changhongedu.cn
[DEBUG] Setting up to push layers : 20.982 ms
[DEBUG] Building resources layer built sha256:788b04502b429b3ef038c576e7d3b5ae90a6e4e011d3a89509035e5b159e59bc
[DEBUG] TIMED   Building resources layer : 677.315 ms
[DEBUG] Building resources layer : 677.315 ms
[DEBUG] Building classes layer built sha256:a944f4f3c341748fd27b9f5c0533e8e87492afc49f6ca4c92a82584799a3080d
[DEBUG] TIMED   Building classes layer : 702.675 ms
[DEBUG] Building classes layer : 702.675 ms
[DEBUG] Building snapshot dependencies layer built sha256:7d07441a36bfd1fcae53adeea9aeb282d603b12d9ce18d0e4c54b49476979565
[DEBUG] TIMED   Building snapshot dependencies layer : 717.751 ms
[DEBUG] Building snapshot dependencies layer : 717.751 ms
[DEBUG] TIMED   Authenticating with push to private-docker.changhongedu.cn : 1106.76 ms
[DEBUG] Authenticating with push to private-docker.changhongedu.cn : 1106.76 ms
[DEBUG] TIMING  Pushing BLOB digest: sha256:a944f4f3c341748fd27b9f5c0533e8e87492afc49f6ca4c92a82584799a3080d, size: 6284
[DEBUG] TIMING  Pushing BLOB digest: sha256:788b04502b429b3ef038c576e7d3b5ae90a6e4e011d3a89509035e5b159e59bc, size: 1159
[DEBUG] TIMING  Pushing BLOB digest: sha256:7d07441a36bfd1fcae53adeea9aeb282d603b12d9ce18d0e4c54b49476979565, size: 288136
[DEBUG] RUNNING Pushing BLOB digest: sha256:788b04502b429b3ef038c576e7d3b5ae90a6e4e011d3a89509035e5b159e59bc, size: 1159
[DEBUG] RUNNING Pushing BLOB digest: sha256:a944f4f3c341748fd27b9f5c0533e8e87492afc49f6ca4c92a82584799a3080d, size: 6284
[DEBUG] RUNNING Pushing BLOB digest: sha256:7d07441a36bfd1fcae53adeea9aeb282d603b12d9ce18d0e4c54b49476979565, size: 288136
[INFO] The base image requires auth. Trying again for hub.c.163.com/azul/zulu-openjdk:8...
[INFO] Retrieving registry credentials for hub.c.163.com...
[DEBUG] TIMING  Retrieving registry credentials for hub.c.163.com
[DEBUG] RUNNING Retrieving registry credentials for hub.c.163.com
[DEBUG] Using Maven settings for hub.c.163.com
[DEBUG] TIMED   Retrieving registry credentials for hub.c.163.com : 5.562 ms
[DEBUG] Retrieving registry credentials for hub.c.163.com : 5.562 ms
[DEBUG] BLOB : digest: sha256:a944f4f3c341748fd27b9f5c0533e8e87492afc49f6ca4c92a82584799a3080d, size: 6284 already exists on registry
[DEBUG] BLOB : digest: sha256:7d07441a36bfd1fcae53adeea9aeb282d603b12d9ce18d0e4c54b49476979565, size: 288136 already exists on registry
[DEBUG] BLOB : digest: sha256:788b04502b429b3ef038c576e7d3b5ae90a6e4e011d3a89509035e5b159e59bc, size: 1159 already exists on registry
[DEBUG] TIMED   Pushing BLOB digest: sha256:7d07441a36bfd1fcae53adeea9aeb282d603b12d9ce18d0e4c54b49476979565, size: 288136 : 78.789 ms
[DEBUG] TIMED   Pushing BLOB digest: sha256:a944f4f3c341748fd27b9f5c0533e8e87492afc49f6ca4c92a82584799a3080d, size: 6284 : 74.492 ms
[DEBUG] Pushing BLOB digest: sha256:7d07441a36bfd1fcae53adeea9aeb282d603b12d9ce18d0e4c54b49476979565, size: 288136 : 78.789 ms
[DEBUG] TIMED   Pushing BLOB digest: sha256:788b04502b429b3ef038c576e7d3b5ae90a6e4e011d3a89509035e5b159e59bc, size: 1159 : 82.856 ms
[DEBUG] Pushing BLOB digest: sha256:a944f4f3c341748fd27b9f5c0533e8e87492afc49f6ca4c92a82584799a3080d, size: 6284 : 74.492 ms
[DEBUG] Pushing BLOB digest: sha256:788b04502b429b3ef038c576e7d3b5ae90a6e4e011d3a89509035e5b159e59bc, size: 1159 : 82.856 ms
[DEBUG] TIMED   Pulling base image manifest : 2045.515 ms
[DEBUG] Pulling base image manifest : 2045.515 ms
[DEBUG] TIMED   Building and pushing image : 2307.967 ms
[DEBUG] Building and pushing image : 2307.967 ms
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 8.041 s
[INFO] Finished at: 2018-09-06T09:59:16+08:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.google.cloud.tools:jib-maven-plugin:0.9.10:build (default-cli) on project product-upgrade-server: Build image failed, perhaps you should make sure your credentials for 'private-docker.changhongedu.cn' are set up correctly: Unauthorized for private-docker.changhongedu.cn/pr
oduct-upgrade: 401 Unauthorized
[ERROR] Auth failed.
[ERROR] -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal com.google.cloud.tools:jib-maven-plugin:0.9.10:build (default-cli) on project product-upgrade-server: Build image failed, perhaps you should make sure your credentials for 'private-docker.changhongedu.cn' are set up correctly
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:213)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:954)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
Caused by: org.apache.maven.plugin.MojoExecutionException: Build image failed, perhaps you should make sure your credentials for 'private-docker.changhongedu.cn' are set up correctly
    at com.google.cloud.tools.jib.maven.BuildImageMojo.execute (BuildImageMojo.java:140)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:208)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:954)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
Caused by: com.google.cloud.tools.jib.registry.RegistryUnauthorizedException: Unauthorized for private-docker.changhongedu.cn/product-upgrade
    at com.google.cloud.tools.jib.plugins.common.BuildStepsRunner.build (BuildStepsRunner.java:206)
    at com.google.cloud.tools.jib.maven.BuildImageMojo.execute (BuildImageMojo.java:136)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:208)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:954)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
Caused by: com.google.api.client.http.HttpResponseException: 401 Unauthorized
Auth failed.

    at com.google.api.client.http.HttpRequest.execute (HttpRequest.java:1070)
    at com.google.cloud.tools.jib.http.Connection.send (Connection.java:161)
    at com.google.cloud.tools.jib.http.Connection.get (Connection.java:116)
    at com.google.cloud.tools.jib.registry.RegistryAuthenticator.authenticate (RegistryAuthenticator.java:253)
    at com.google.cloud.tools.jib.registry.RegistryAuthenticator.authenticatePull (RegistryAuthenticator.java:216)
    at com.google.cloud.tools.jib.builder.steps.PullBaseImageStep.call (PullBaseImageStep.java:162)
    at com.google.cloud.tools.jib.builder.steps.PullBaseImageStep.call (PullBaseImageStep.java:57)
    at com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.runInterruptibly (TrustedListenableFutureTask.java:127)
    at com.google.common.util.concurrent.InterruptibleTask.run (InterruptibleTask.java:57)
    at com.google.common.util.concurrent.TrustedListenableFutureTask.run (TrustedListenableFutureTask.java:80)
    at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1149)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:624)
    at java.lang.Thread.run (Thread.java:748)
[ERROR]
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

**Additional Information**: <!-- Any additional information that may be helpful -->
The authorization for 'private-docker.changhongedu.cn' is correct.
'hub.c.163.com' is just a mirror for 'hub.docker.com' and the authorization of 'hub.docker.com' can not be used to 'hub.c.163.com' , so the jib:build report a "401 Unauthorized" message, but the stack trace message give me a misleading to 'private-docker.changhongedu.cn', it should be 'Unauthorized for hub.c.163.com/azul/zulu-openjdk:8', right?

<!-- Thanks for contributing! -->

--- a/jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryAuthenticationFailedException.java
+++ b/jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryAuthenticationFailedException.java
@@ -16,21 +16,23 @@
 
 package com.google.cloud.tools.jib.registry;
 
+import java.text.MessageFormat;
+
 /** Thrown because registry authentication failed. */
 public class RegistryAuthenticationFailedException extends Exception {
 
-  private static final String REASON_PREFIX = "Failed to authenticate with the registry because: ";
+  private static final String REASON = "Failed to authenticate with registry {0}/{1} because: {2}";
   private final String serverUrl;
   private final String imageName;
 
   RegistryAuthenticationFailedException(String serverUrl, String imageName, Throwable cause) {
-    super(REASON_PREFIX + cause.getMessage(), cause);
+    super(MessageFormat.format(REASON, serverUrl, imageName, cause.getMessage()), cause);
     this.serverUrl = serverUrl;
     this.imageName = imageName;
   }
 
   RegistryAuthenticationFailedException(String serverUrl, String imageName, String reason) {
-    super(REASON_PREFIX + reason);
+    super(MessageFormat.format(REASON, serverUrl, imageName, reason));
     this.serverUrl = serverUrl;
     this.imageName = imageName;
   }

--- a/jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryAuthenticationFailedException.java
+++ b/jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryAuthenticationFailedException.java
@@ -20,12 +20,28 @@ package com.google.cloud.tools.jib.registry;
 public class RegistryAuthenticationFailedException extends Exception {
 
   private static final String REASON_PREFIX = "Failed to authenticate with the registry because: ";
+  private final String serverUrl;
+  private final String imageName;
 
-  RegistryAuthenticationFailedException(Throwable cause) {
+  RegistryAuthenticationFailedException(String serverUrl, String imageName, Throwable cause) {
     super(REASON_PREFIX + cause.getMessage(), cause);
+    this.serverUrl = serverUrl;
+    this.imageName = imageName;
   }
 
-  RegistryAuthenticationFailedException(String reason) {
+  RegistryAuthenticationFailedException(String serverUrl, String imageName, String reason) {
     super(REASON_PREFIX + reason);
+    this.serverUrl = serverUrl;
+    this.imageName = imageName;
+  }
+
+  /** @return the server being authenticated */
+  public String getServerUrl() {
+    return serverUrl;
+  }
+
+  /** @return the image being authenticated */
+  public String getImageName() {
+    return imageName;
   }
 }

--- a/jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryAuthenticator.java
+++ b/jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryAuthenticator.java
@@ -87,7 +87,7 @@ public class RegistryAuthenticator {
             .getRegistryAuthenticator();
 
       } catch (MalformedURLException ex) {
-        throw new RegistryAuthenticationFailedException(ex);
+        throw new RegistryAuthenticationFailedException(serverUrl, repository, ex);
 
       } catch (InsecureRegistryException ex) {
         // Cannot skip certificate validation or use HTTP, so just return null.
@@ -133,13 +133,21 @@ public class RegistryAuthenticator {
 
     // Checks that the authentication method starts with 'bearer ' (case insensitive).
     if (!authenticationMethod.matches("^(?i)(bearer) .*")) {
-      throw newRegistryAuthenticationFailedException(authenticationMethod, "Bearer");
+      throw newRegistryAuthenticationFailedException(
+          registryEndpointRequestProperties.getServerUrl(),
+          registryEndpointRequestProperties.getImageName(),
+          authenticationMethod,
+          "Bearer");
     }
 
     Pattern realmPattern = Pattern.compile("realm=\"(.*?)\"");
     Matcher realmMatcher = realmPattern.matcher(authenticationMethod);
     if (!realmMatcher.find()) {
-      throw newRegistryAuthenticationFailedException(authenticationMethod, "realm");
+      throw newRegistryAuthenticationFailedException(
+          registryEndpointRequestProperties.getServerUrl(),
+          registryEndpointRequestProperties.getImageName(),
+          authenticationMethod,
+          "realm");
     }
     String realm = realmMatcher.group(1);
 
@@ -151,13 +159,14 @@ public class RegistryAuthenticator {
             ? serviceMatcher.group(1)
             : registryEndpointRequestProperties.getServerUrl();
 
-    return new RegistryAuthenticator(
-        realm, service, registryEndpointRequestProperties.getImageName());
+    return new RegistryAuthenticator(realm, service, registryEndpointRequestProperties);
   }
 
   private static RegistryAuthenticationFailedException newRegistryAuthenticationFailedException(
-      String authenticationMethod, String authParam) {
+      String registry, String repository, String authenticationMethod, String authParam) {
     return new RegistryAuthenticationFailedException(
+        registry,
+        repository,
         "'"
             + authParam
             + "' was not found in the 'WWW-Authenticate' header, tried to parse: "
@@ -189,10 +198,21 @@ public class RegistryAuthenticator {
   }
 
   private final String authenticationUrlBase;
+  private final RegistryEndpointRequestProperties registryEndpointRequestProperties;
   @Nullable private Authorization authorization;
 
-  RegistryAuthenticator(String realm, String service, String repository) {
-    authenticationUrlBase = realm + "?service=" + service + "&scope=repository:" + repository + ":";
+  RegistryAuthenticator(
+      String realm,
+      String service,
+      RegistryEndpointRequestProperties registryEndpointRequestProperties) {
+    authenticationUrlBase =
+        realm
+            + "?service="
+            + service
+            + "&scope=repository:"
+            + registryEndpointRequestProperties.getImageName()
+            + ":";
+    this.registryEndpointRequestProperties = registryEndpointRequestProperties;
   }
 
   /**
@@ -257,13 +277,18 @@ public class RegistryAuthenticator {
             JsonTemplateMapper.readJson(responseString, AuthenticationResponseTemplate.class);
         if (responseJson.getToken() == null) {
           throw new RegistryAuthenticationFailedException(
+              registryEndpointRequestProperties.getServerUrl(),
+              registryEndpointRequestProperties.getImageName(),
               "Did not get token in authentication response from " + authenticationUrl);
         }
         return Authorizations.withBearerToken(responseJson.getToken());
       }
 
     } catch (IOException ex) {
-      throw new RegistryAuthenticationFailedException(ex);
+      throw new RegistryAuthenticationFailedException(
+          registryEndpointRequestProperties.getServerUrl(),
+          registryEndpointRequestProperties.getImageName(),
+          ex);
     }
   }
 }

--- a/jib-core/src/test/java/com/google/cloud/tools/jib/registry/RegistryAuthenticationFailedExceptionTest.java
+++ b/jib-core/src/test/java/com/google/cloud/tools/jib/registry/RegistryAuthenticationFailedExceptionTest.java
@@ -29,7 +29,8 @@ public class RegistryAuthenticationFailedExceptionTest {
     Assert.assertEquals("serverUrl", exception.getServerUrl());
     Assert.assertEquals("imageName", exception.getImageName());
     Assert.assertEquals(
-        "Failed to authenticate with the registry because: message", exception.getMessage());
+        "Failed to authenticate with registry serverUrl/imageName because: message",
+        exception.getMessage());
   }
 
   @Test
@@ -41,6 +42,7 @@ public class RegistryAuthenticationFailedExceptionTest {
     Assert.assertEquals("imageName", exception.getImageName());
     Assert.assertSame(cause, exception.getCause());
     Assert.assertEquals(
-        "Failed to authenticate with the registry because: message", exception.getMessage());
+        "Failed to authenticate with registry serverUrl/imageName because: message",
+        exception.getMessage());
   }
 }

--- a/jib-core/src/test/java/com/google/cloud/tools/jib/registry/RegistryAuthenticationFailedExceptionTest.java
+++ b/jib-core/src/test/java/com/google/cloud/tools/jib/registry/RegistryAuthenticationFailedExceptionTest.java
@@ -1,0 +1,46 @@
+/*
+ * Copyright 2018 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.jib.registry;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+/** Tests for {@link RegistryAuthenticationFailedException}. */
+public class RegistryAuthenticationFailedExceptionTest {
+
+  @Test
+  public void testRegistryAuthenticationFailedException_message() {
+    RegistryAuthenticationFailedException exception =
+        new RegistryAuthenticationFailedException("serverUrl", "imageName", "message");
+    Assert.assertEquals("serverUrl", exception.getServerUrl());
+    Assert.assertEquals("imageName", exception.getImageName());
+    Assert.assertEquals(
+        "Failed to authenticate with the registry because: message", exception.getMessage());
+  }
+
+  @Test
+  public void testRegistryAuthenticationFailedException_exception() {
+    Throwable cause = new Throwable("message");
+    RegistryAuthenticationFailedException exception =
+        new RegistryAuthenticationFailedException("serverUrl", "imageName", cause);
+    Assert.assertEquals("serverUrl", exception.getServerUrl());
+    Assert.assertEquals("imageName", exception.getImageName());
+    Assert.assertSame(cause, exception.getCause());
+    Assert.assertEquals(
+        "Failed to authenticate with the registry because: message", exception.getMessage());
+  }
+}

--- a/jib-core/src/test/java/com/google/cloud/tools/jib/registry/RegistryAuthenticatorTest.java
+++ b/jib-core/src/test/java/com/google/cloud/tools/jib/registry/RegistryAuthenticatorTest.java
@@ -74,7 +74,7 @@ public class RegistryAuthenticatorTest {
 
     } catch (RegistryAuthenticationFailedException ex) {
       Assert.assertEquals(
-          "Failed to authenticate with the registry because: 'Bearer' was not found in the 'WWW-Authenticate' header, tried to parse: realm=\"https://somerealm\",service=\"someservice\",scope=\"somescope\"",
+          "Failed to authenticate with registry someserver/someimage because: 'Bearer' was not found in the 'WWW-Authenticate' header, tried to parse: realm=\"https://somerealm\",service=\"someservice\",scope=\"somescope\"",
           ex.getMessage());
     }
   }
@@ -88,7 +88,7 @@ public class RegistryAuthenticatorTest {
 
     } catch (RegistryAuthenticationFailedException ex) {
       Assert.assertEquals(
-          "Failed to authenticate with the registry because: 'realm' was not found in the 'WWW-Authenticate' header, tried to parse: Bearer scope=\"somescope\"",
+          "Failed to authenticate with registry someserver/someimage because: 'realm' was not found in the 'WWW-Authenticate' header, tried to parse: Bearer scope=\"somescope\"",
           ex.getMessage());
     }
   }

--- a/jib-plugins-common/src/main/java/com/google/cloud/tools/jib/plugins/common/BuildStepsRunner.java
+++ b/jib-plugins-common/src/main/java/com/google/cloud/tools/jib/plugins/common/BuildStepsRunner.java
@@ -199,10 +199,12 @@ public class BuildStepsRunner {
 
       } else if (exceptionDuringBuildSteps instanceof RegistryAuthenticationFailedException
           && exceptionDuringBuildSteps.getCause() instanceof HttpResponseException) {
+        RegistryAuthenticationFailedException failureException =
+            (RegistryAuthenticationFailedException) exceptionDuringBuildSteps;
         handleRegistryUnauthorizedException(
             new RegistryUnauthorizedException(
-                buildConfiguration.getTargetImageConfiguration().getImageRegistry(),
-                buildConfiguration.getTargetImageConfiguration().getImageRepository(),
+                failureException.getServerUrl(),
+                failureException.getImageName(),
                 (HttpResponseException) exceptionDuringBuildSteps.getCause()),
             helpfulSuggestions);
 

